packages/react-native-web/src/exports/Image/index.js

```diff
@@ -257,21 +256,23 @@ const Image: React.AbstractComponent<
 
   // Image loading
   const uri = resolveAssetUri(source);
-  React.useEffect(() => {
-    abortPendingRequest();
+  let headers;
+  if (source && typeof source.headers === 'object') {
+    headers = ((source.headers: any): { [key: string]: string });
+  }
```

Comment:
These are extracted here so that we trigger/re-trigger `load` only if `uri` or `headers` change

The weird syntax gymnastics are flowtype casts to satisfy the type checker 
Perhaps someone can propose a cleaner alternative or an ignore annotation